airbyte-integrations/connectors/destination-customer-io/src/test/kotlin/io/airbyte/integrations/destination/customerio/batch/DeclarativeBatchEntryAssemblerTest.kt

```diff
@@ -0,0 +1,66 @@
+package io.airbyte.integrations.destination.customerio.batch
+
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.cdk.load.message.DestinationRecordRaw
+import io.airbyte.cdk.load.message.DestinationRecordSource
+import io.airbyte.cdk.util.Jsons
+import io.airbyte.integrations.destination.customerio.io.airbyte.integrations.destination.customerio.batch.DeclarativeBatchEntryAssembler
+import io.mockk.every
+import io.mockk.mockk
+import java.util.UUID
+import kotlin.test.assertEquals
+import org.junit.jupiter.api.Test
+
+class DeclarativeBatchEntryAssemblerTest {
+    val PERSON_EVENT_TEMPLATE = """
+{
+  "type": "person",
+  "identifiers": {
+    "id": "{{ record["person_email"] }}"
+  },
+  "action": "identify",
+  "attributes": {{ additional_properties }}
```

Comment:
I'm not a big fan of passing the whole dictionary here because I'm not sure what we will do in the following cases:
* The additional properties are sent through multiple fields (in this case, we would need to have a logic to split them which I assume is an additive change but I fear it might break the interpolation if we remove `additional_properties` from the interpolation context.
* The API has some rules through which the additional properties need to be passed. For example, if values need to be strings, we don't have anything to enforce that

If you have other idea why it might be a bad idea, please add them to the list